src/main/java/com/dougnoel/sentinel/files/CsvFile.java

```diff
@@ -0,0 +1,334 @@
+        super();
+        csvFormat = CSVFormat.DEFAULT;
+        numHeaderRows = numberOfHeaderRows;
+    }
+
+    /**
+     * Create a CSV file configured with the given number of header rows from the CSV file at the given path.
+     * @param pathToFile Path path to the CSV file.
+     * @param numberOfHeaderRows int number of header rows in the CSV file.
+     * @throws FileNotFoundException in the case that a file does not exist at that path.
+     */
+    public CsvFile(Path pathToFile, int numberOfHeaderRows) throws FileNotFoundException {
+        super(pathToFile);
+        csvFormat = CSVFormat.DEFAULT;
+        numHeaderRows = numberOfHeaderRows;
+    }
+
+    /**
+     * Create a CSV file configured with the given number of header rows and CSVFormat from the CSV file at the given path.
+     * @param pathToFile Path path to the CSV file.
+     * @param format CSVFormat the format of the CSV file. See CSVFormat documentation for more info.
+     * @param numberOfHeaderRows int number of header rows in the CSV file.
+     * @throws FileNotFoundException in the case that a file does not exist at that path.
+     */
+    public CsvFile(Path pathToFile, CSVFormat format, int numberOfHeaderRows) throws FileNotFoundException {
+        super(pathToFile);
+        csvFormat = format;
+        numHeaderRows = numberOfHeaderRows;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        if (!super.equals(o)) return false;
+        CsvFile csvFile = (CsvFile) o;
+        return numHeaderRows == csvFile.numHeaderRows && Objects.equals(csvFormat, csvFile.csvFormat);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(super.hashCode(), csvFormat, numHeaderRows);
+    }
+
+    private CSVParser getParser() throws IOException {
+        return CSVParser.parse(toPath(), Charset.defaultCharset(), csvFormat);
+    }
+
+    /**
+     * Returns the number of rows in the CSV file.
+     * @return int the number of rows in the CSV file.
+     */
+    public int getNumberOfRows(){
+        try(var parser = getParser()){
+            return parser.getRecords().size();
+        }
+        catch(IOException ioe){
+            log.trace(SentinelStringUtils.format("IOException caught while parsing CSV file {}.", toPath()));
+            return -1;
+        }
+    }
+
+    /**
+     * Returns a List&lt;String&gt; consisting of each value in a given row.
+     * The row index starts at 1 and factors out the header rows, so that calling this method with rowIndex=1 gets the first row with actual data.
+     * @param rowIndex int index of the first row with real data, starting at 1, taking into consideration row headers.
+     * @return List&lt;String&gt; consisting of each value in a given row.
+     */
+    public List<String> getRowData(int rowIndex){
```

Comment:
Let's edit all these to "read" so it's clear we are reading the file every time.
```suggestion
    public List<String> readRowData(int rowIndex){
```